### ISSUE TITLE
0.10 FTBFS on s390x, ia64, mips

### ISSUE THREAD

**Issue by infinity0** [cpp-netlib/cpp-netlib#329]
0.10 FTBFS (fails to build from source) on platforms s390x and ia64. I don't have access to the build machines nor an architecture equivalent to debug this properly, but I think I've found the cause.

See build logs for [ia64](https://buildd.debian.org/status/fetch.php?pkg=cpp-netlib&arch=ia64&ver=0.10.1-1&stamp=1385292009) and [s390x](https://buildd.debian.org/status/fetch.php?pkg=cpp-netlib&arch=s390x&ver=0.10.1-1&stamp=1385288228) and [mips](https://buildd.debian.org/status/fetch.php?pkg=cpp-netlib&arch=mips&ver=0.10.1-1&stamp=1385355483), both with the following error:

~~~~
/usr/bin/c++   -g -O2 -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2  -Wall -O2 -g -DNDEBUG    CMakeFiles/cpp-netlib-utils_thread_pool.dir/utils_thread_pool.cpp.o  -o ../../../tests/cpp-netlib-utils_thread_pool -rdynamic -Wl,-Bstatic -lboost_unit_test_framework -lboost_system -lboost_regex -lboost_date_time -lboost_thread -lboost_filesystem -lboost_program_options -lboost_chrono -Wl,-Bdynamic -lpthread -lpthread ../src/libcppnetlib-uri.so.0.10.1 -lssl -lcrypto -lrt -Wl,-rpath,/«PKGBUILDDIR»/obj-ia64-linux-gnu/libs/network/src 
/usr/lib/gcc/ia64-linux-gnu/4.6/../../../libboost_thread.a(once.o): In function `boost::thread_detail::commit_once_region(boost::once_flag&)':
(.text+0xd2): undefined reference to `boost::atomics::detail::lockpool::get_lock_for(void const volatile*)'
[..]
collect2: ld returned 1 exit status
make[3]: *** [tests/cpp-netlib-utils_thread_pool] Error 1
~~~~

A previous issue has already been fixed Boost upstream and [in Debian](http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=721577). However as the above build logs show, they are already using the fixed versions of Boost (1.54.0-3), so there is something else going on here.

I believe the "something else" is over-use of the `-Bstatic` flag by cmake. From `man ld`:

~~~~
       -Bstatic
       -dn
       -non_shared
       -static
           Do not link against shared libraries.  This is only meaningful on platforms for which shared libraries are
           supported.  The different variants of this option are for compatibility with various systems.  You may use
           this option multiple times on the command line: it affects library searching for -l options which follow
           it.  This option also implies --unresolved-symbols=report-all.  This option can be used with -shared.
           Doing so means that a shared library is being created but that all of the library's external references
           must be resolved by pulling in entries from static libraries.
~~~~

In other words, it links against libboost_thread.a rather than .so, effectively meaning that dependencies are not resolved transitively. To fix this, cpp-netlib will need to explicitly link against libboost_atomic.a on the appropriate platforms (not ideal, since this is an implementation detail of libboost_thread that might change), or tell cmake to drop the `-Bstatic` flag and link against libboost_thread.so instead.
